File: replace_roles.py
import os, glob, re

path = 'c:/Users/GANES/Downloads/AI-Dairy-Farm-Management/AI-Dairy-Farm-Management/backend/src/main/java/com/dairycare/controller/*.java'
for file in glob.glob(path):
    with open(file, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # We only want to touch PreAuthorize annotations that contain USER or FARMER to avoid breaking others
    # but wait, the prompt says "Dairy Farm Owner (Admin)". 
    # Let's just blindly replace all PreAuthorize to ADMIN in these specific controllers.
    # Actually, some might be public. If they have PreAuthorize, they should be ADMIN.
    
    if "AuthController" in file or "DairyProduct" in file or "ProductSale" in file:
        continue # Skip auth and storefront controllers
        
    new_content = re.sub(r'@PreAuthorize\([^)]+\)', '@PreAuthorize("hasRole(\'ADMIN\')")', content)
    
    if new_content != content:
        with open(file, 'w', encoding='utf-8') as f:
            f.write(new_content)
        print(f'Updated {os.path.basename(file)}')

File: replace_roles.js
const fs = require('fs');
const path = require('path');

const dir = 'c:/Users/GANES/Downloads/AI-Dairy-Farm-Management/AI-Dairy-Farm-Management/backend/src/main/java/com/dairycare/controller';
const files = fs.readdirSync(dir);

files.forEach(file => {
    if (file === 'AuthController.java' || file === 'DairyProductController.java' || file === 'ProductSaleController.java' || file === 'UserController.java') {
        return; // skip
    }
    
    const filePath = path.join(dir, file);
    let content = fs.readFileSync(filePath, 'utf-8');
    
    // Replace @PreAuthorize("hasRole(...)") with @PreAuthorize("hasRole('ADMIN')")
    const newContent = content.replace(/@PreAuthorize\([^)]+\)/g, '@PreAuthorize("hasRole(\'ADMIN\')")');
    
    if (content !== newContent) {
        fs.writeFileSync(filePath, newContent, 'utf-8');
        console.log(`Updated ${file}`);
    }
});
